# Generated by Django 4.2.14 on 2024-08-18 10:11

from django.conf import settings
import django.core.validators
from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    dependencies = [
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
        ('api', '0007_problemtest'),
    ]

    operations = [
        migrations.CreateModel(
            name='Submission',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('user_code', models.TextField()),
                ('language', models.CharField(choices=[('python', 'Python'), ('java', 'Java'), ('cpp', 'C++'), ('javascript', 'JavaScript')], max_length=10)),
                ('created_at', models.DateTimeField(auto_now_add=True)),
                ('status', models.CharField(choices=[('Failed', 'Failed'), ('Pending', 'Pending'), ('Completed', 'Completed'), ('Time Limit Exceeded', 'Time Limit Exceeded'), ('Runtime Error', 'Runtime Error'), ('Compilation Error', 'Compilation Error')], max_length=20)),
                ('total_score', models.PositiveIntegerField(default=0, validators=[django.core.validators.MinValueValidator(0), django.core.validators.MaxValueValidator(100)])),
                ('problem', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='api.problem')),
                ('user', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to=settings.AUTH_USER_MODEL)),
            ],
        ),
        migrations.CreateModel(
            name='UserProblemScore',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('score', models.PositiveIntegerField(default=0, validators=[django.core.validators.MinValueValidator(0), django.core.validators.MaxValueValidator(100)])),
                ('problem', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='api.problem')),
                ('user', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to=settings.AUTH_USER_MODEL)),
            ],
        ),
        migrations.CreateModel(
            name='SubmissionTest',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('status', models.CharField(choices=[('Accepted', 'Accepted'), ('Wrong Answer', 'Wrong Answer'), ('Time Limit Exceeded', 'Time Limit Exceeded'), ('Runtime Error', 'Runtime Error'), ('Compilation Error', 'Compilation Error')], max_length=20)),
                ('score', models.PositiveIntegerField(default=0, validators=[django.core.validators.MinValueValidator(0), django.core.validators.MaxValueValidator(100)])),
                ('created_at', models.DateTimeField(auto_now_add=True)),
                ('problem_test', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='api.problemtest')),
                ('submission', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='api.submission')),
            ],
        ),
    ]
